docs/coding-conventions.md

```diff
@@ -0,0 +1,297 @@
+    } else {
+        return doAnotherThing()
+    }
+} else {
+    return err
+}
+```
+
+### Good example
+
+```go
+val, err := doSomething()
+if err != nil {
+    return err
+}
+
+if val {
+    return doSomethingElse()
+}
+
+return doAnotherThing()
+```
+
+## Use the same coding style throughout a file
+
+- When adding to a file, stick to the existing coding style to make reviews
+  easier.
+- When you encounter code that requires a larger cleanup do it in a separate
+  commit or PR.
+
+### Bad example
+
+```go
+func checkAB(in string) bool {
+    var bool ok
+    if in == "A" || in == "B" {
+        ok = true
+    } else if in == "C" {
+        ok = false
+    }
+    return ok
+}
+
+func checkBC(in string) bool {
+    switch in {
+    case "B", "C":
+        return true
+    }
+
+    return false
+}
+```
+
+### Good example
+
+```go
+func checkAB(in string) bool {
+    return in == "A" || in == "B"
+}
+
+func checkBC(in string) bool {
+    return in == "B" || in == "C"
+}
+```
+
+# Avoid repetition of hardcoded values
+
+- Use constants if you would repeat hardcoded values otherwise.
+- Keep them private if they are used only in a single package.
+- Carefully consider if you should make a constant exported.
+
+### Bad example
+
+```go
+func getImportantAnnotation(obj metav1.ObjectMeta) string {
+    return obj.GetAnnotations()["kubevirt.io/my-annotation"]
+}
+
+func setImportantAnnotation(obj metav1.ObjectMeta, val string) {
+    obj.GetAnnotations()["kubevirt.io/my-annotation"] = val
+}
+```
+
+### Good example
+
+```go
+const annotationKey = "kubevirt.io/my-annotation"
+
+func getImportantAnnotation(obj metav1.ObjectMeta) string {
+    return obj.GetAnnotations()[annotationKey]
+}
+
+func setImportantAnnotation(obj metav1.ObjectMeta, val string) {
+    obj.GetAnnotations()[annotationKey] = val
+}
+```
+
+# Uniform import order and naming
+
+- Use the following import order with one block per item:
+    - Golang standard libraries
+    - Ginkgo / Gomega imports (only in test files)
+    - Third-party libraries
+    - Local packages
+- Use the following naming schemes for imports:
+    - `virtv1` for imports of `kubevirt.io/api`
+    - `metav1` for imports of `k8s.io/apimachinery/pkg/apis/meta/v1`
+    - `k8s` prefix for `k8s.io` imports on collision with other imports
+    - TODO: Add more
+- `make format` is able to help you with this.
+
+### Bad example
+
+```go
+import (
+    "context"
+    k8sv1 "k8s.io/api/core/v1"
+    "time"
+
+    ."github.com/onsi/ginkgo/v2"
+    ."github.com/onsi/gomega"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    v1 "kubevirt.io/api/core/v1"
+    "kubevirt.io/kubevirt/tests/clientcmd"
+
+    "kubevirt.io/kubevirt/tests/testsuite"
+)
+```
+
+### Good example
+
+```go
+import (
+    "context"
+    "time"
+
+    ."github.com/onsi/ginkgo/v2"
+    ."github.com/onsi/gomega"
+
+    k8sv1 "k8s.io/api/core/v1"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+    virtv1 "kubevirt.io/api/core/v1"
```

Comment:
`kubevirt.io/api/core/v1` is not a local package per se, it is because it is replaced with `./staging/src/kubevirt.io/api` in `go.mod`. But I would argue from the looks it is an external dependency.

Reply:
Ok, then how about this suggestion:

> Attempt to group the packages by domain when possible.

In this case, the `kubevirt.io` is a different domain from `k8s.io`.

I also do not group it with the `kubevirt.io/kubevirt` ones, I just place it in its own block.

Reply:
Agree, updated it.